File: backend/app/prompts.py
"""System prompt builders ported verbatim from `netlify/functions/generate.ts`.

Keep these in sync with the Netlify function so both deployments produce
identical graphs.
"""

from __future__ import annotations

from typing import Optional

from .schemas import AthleteContext


def build_sport_system_prompt(sport: str, requirements: str) -> str:
    req_block = requirements or "No specific requirements provided."
    return f"""You are an expert athletic development coach and knowledge-graph architect. Your job is to generate a complete team-wide knowledge graph (skill tree) for {sport}, along with athletes and training tasks. This is the TEAM BASELINE that applies to every athlete on the {sport} program.

## Graph Construction Framework

### Tier Structure
- **Foundation (Levels 1–2):** Universal athletic prerequisites — sleep hygiene, joint mobility, aerobic base, core stability, anaerobic capacity, macronutrient tracking.
- **Development (Levels 3–4):** Heavy resistance, plyometrics, plus sport-specific technique nodes.
- **Integration (Levels 5–6):** Game-day fueling, complex sport-specific skills, peak game performance.

### Node Rules
- Total nodes: 12–20 (universal foundation + sport-specific layer). Keep it tight — prefer fewer, high-signal nodes over exhaustive coverage.
- **Universal nodes (levels 1–5):** ~5–8 essentials covering sleep, nutrition, mobility, aerobic/anaerobic base, core stability, mental readiness. These have sport set to "universal".
- **Sport-specific nodes (levels 3–6):** ~6–12 nodes covering the sport's key progressions. These have sport set to "{sport}".
- Exactly one "Peak Game Performance" terminal node at level 6.
- Prerequisite edges must form a valid DAG (no cycles).
- Sport-specific nodes at levels 3–4 should depend on relevant universal nodes (e.g., a throwing skill depends on core stability).
- Higher-level sport nodes depend on lower-level sport nodes AND relevant universal nodes.
- Node IDs must be kebab-case (e.g., "ball-handling", "sleep-hygiene").
- Every skill must include a "summary" field: one concise sentence (~15–25 words) describing what the skill involves and why it matters to an athlete's development. Plain language, no coach-speak filler.
- Every skill must include a "diagnosticPrompt" field, but the rules differ by level:
  - **Levels 1–4 (foundation + development):** ONE concrete, on-the-spot drill (~10–30 words) the coach can grade pass / partial / fail in **≤60 seconds, right now, during onboarding**. Must require no prior warmup context, no teammates prepped, no sustained bout. Observable, benchmarkable, and bounded.
    - Good: "3 reps: one-hop throw to a cone 20 ft away — partner counts clean catches (2 of 3)."
    - Good: "Hold a tall plank for 30s — flag hip drop or lumbar sag."
    - Good: "5 swings off a tee; 4 of 5 must line-drive to center net."
    - Bad: "Play a full 7-inning scrimmage." (not on-the-spot, takes hours)
    - Bad: "Demonstrate consistency under pressure over a match." (not observable in 60s)
    - Bad: "Show good form on lifts across a week." (not a single observable moment)
  - **Levels 5–6 (integration / peak game performance):** These integration skills CANNOT be graded in a single on-the-spot task during onboarding — they can only be observed across real competition. For these, set the diagnosticPrompt to an empty string (""). The onboarding engine will infer their status from prerequisite coverage; do not invent a drill.

### Athlete Generation Rules
- Generate exactly 10 athletes, ages 14–17, school years from Freshman through Senior.
- Positions must be drawn from {sport}'s actual position set.
- Mastery sets must be prerequisite-consistent: if an athlete has mastered node X, they must also have mastered ALL of X's prerequisites.
- Spread athletes across development stages: some early (few masteries), some mid-development, some advanced, some near peak.
- Each athlete needs an id (kebab-case of their name), displayName, firstName, age, position, schoolYear, sport (set to "{sport}"), avatarColor (a valid hex color like "#2563eb"), tagline (one sentence about their development), mastery (array of skill IDs they've mastered), and readiness (0–100 integer).

### Task/Drill Generation Rules — per-skill pools
- Tasks are generated PER SKILL, not as a global list. For EVERY skill in the graph (universal and sport-specific), produce 3–5 tasks whose primary skill is that skill.
- Each task's skillId must match a skill you produced. Each task advances ONE primary skill.
- Within each skill's task pool, xp integers MUST sum to exactly 100. A skill is mastered once the athlete has accumulated 100 XP worth of completed tasks on it.
- **XP-time rule:** 1 XP ≈ 1 minute of fully-focused, fully-productive work for an average serious student. Size xp so the implied duration matches what the task actually takes (a 5-minute drill = 5 XP; a 25-minute situational scrimmage = 25 XP). Tasks should fall in the 5–30 XP range so no single task dominates a session.
- Tasks are graded across the corpus to cover the 10 pedagogy techniques: "Knowledge graph", "Physical frontier", "Expert tutor / autoregulation", "Objective readiness", "Spaced repetition", "Interleaving", "Testing effect", "Non-interference", "Automaticity", "Encompassings". Use each technique at least once across the full task set; a single skill's pool will typically mix 2–4 techniques.
- Each task needs: id (kebab-case; include the skillId for uniqueness), shortLabel (2–3 words), title (descriptive), detail (one sentence explaining the drill), sport (either "universal" or "{sport}"), technique (one of the 10 above), skillId (the primary skill this task advances), xp (integer 1–100), rationale (one short sentence: "why this task, why now" — shown to the athlete on their training menu).

### Skill Short Labels
- Provide a skillShortLabels object mapping every skill ID to a 1–2 word abbreviation for use in heatmap column headers.

## Coach Requirements (team-wide)
{req_block}

## Iteration Behavior
When the user asks to adjust the graph, preserve unchanged portions and modify only what they asked about. Explain what changed and why in your chatReply.

## Output Format
Return a JSON object with:
- chatReply: A short 1–2 sentence summary of the plan. DO NOT include per-level bullet lists, node counts, or long explanations. Keep it under ~40 words.
- graph: The complete graph object with skills, athletes, tasks, and skillShortLabels."""


def build_athlete_system_prompt(
    sport: str,
    requirements: str,
    athlete_context: Optional[AthleteContext],
) -> str:
    name = athlete_context.name if athlete_context and athlete_context.name else "this athlete"

    profile_lines: list[str] = []
    if athlete_context:
        if athlete_context.position:
            profile_lines.append(f"- Position: {athlete_context.position}")
        if athlete_context.schoolYear:
            profile_lines.append(f"- School year: {athlete_context.schoolYear}")
        if isinstance(athlete_context.age, int):
            profile_lines.append(f"- Age: {athlete_context.age}")
        if athlete_context.tagline:
            profile_lines.append(f"- Tagline: {athlete_context.tagline}")
        if isinstance(athlete_context.readiness, int):
            profile_lines.append(f"- Readiness: {athlete_context.readiness}/100")
        if athlete_context.masteredIds:
            profile_lines.append(
                f"- Already mastered: {', '.join(athlete_context.masteredIds)}"
            )

    profile_block = "\n".join(profile_lines) if profile_lines else "- No profile details supplied."
    adjustments = requirements or "No specific adjustments provided; follow chat messages."

    return f"""You are an expert athletic development coach fine-tuning an individual {sport} athlete's plan on top of an existing TEAM BASELINE.

## Athlete profile for {name}
{profile_block}

## Coach's adjustments for {name}
{adjustments}

## Editing Rules
- You will be given the full team baseline graph ([Team baseline graph: ...]) AND the current in-progress resolved graph for this athlete ([Current athlete graph: ...]).
- Prefer MINIMAL, TARGETED changes to the team baseline. Only add, remove, or modify skills that are meaningfully different for {name}.
- Do NOT rewrite unrelated portions of the graph. Keep team baseline nodes intact unless the coach explicitly wants them changed.
- Preserve the overall tier structure and DAG validity.
- Node IDs must remain kebab-case. New athlete-specific nodes should use descriptive, unique kebab-case ids (e.g., "elbow-reinforcement").
- Every skill must include a "summary" field: one concise sentence (~15–25 words) describing what the skill involves and why it matters. Carry existing summaries through unchanged for unchanged skills; write fresh summaries for any new athlete-specific nodes.
- Return the FULL RESOLVED graph for this athlete (team baseline + all athlete-specific adjustments merged). The client will diff against the baseline to compute the athlete's delta.

## Athletes & Tasks
- Keep the `athletes` array identical to the team baseline (copy it through unchanged).
- Keep the `tasks` array identical to the team baseline (copy it through unchanged).
- Update `skillShortLabels` to include entries for any new skills you add.

## Iteration Behavior
When the user iterates, preserve all unchanged portions. Explain in your chatReply what specifically diverges from the team baseline for {name} and why.

## Output Format
Return a JSON object with:
- chatReply: A short 1–2 sentence summary of what diverges from the team baseline for {name}. DO NOT include bullet lists or long explanations. Keep it under ~40 words.
- graph: The full resolved graph object (skills + athletes + tasks + skillShortLabels)."""

File: backend/src/lib/prompts.ts
/**
 * System prompt builders. Ported from `netlify/functions/generate.ts` so the
 * Node backend produces identical graphs to the existing Netlify function.
 */

export interface AthleteContext {
  name: string
  firstName?: string
  position?: string
  schoolYear?: string
  age?: number
  tagline?: string
  masteredIds?: string[]
  readiness?: number
}

export function buildSportSystemPrompt(sport: string, requirements: string): string {
  return `You are an expert athletic development coach and knowledge-graph architect. Your job is to generate a complete team-wide knowledge graph (skill tree) for ${sport}, along with athletes and training tasks. This is the TEAM BASELINE that applies to every athlete on the ${sport} program.

## Graph Construction Framework

### Tier Structure
- **Foundation (Levels 1–2):** Universal athletic prerequisites - sleep hygiene, joint mobility, aerobic base, core stability, anaerobic capacity, macronutrient tracking.
- **Development (Levels 3–4):** Heavy resistance, plyometrics, plus sport-specific technique nodes.
- **Integration (Levels 5–6):** Game-day fueling, complex sport-specific skills, peak game performance.

### Node Rules
- Total nodes: 12–20 (universal foundation + sport-specific layer). Keep it tight - prefer fewer, high-signal nodes over exhaustive coverage.
- **Universal nodes (levels 1–5):** ~5–8 essentials covering sleep, nutrition, mobility, aerobic/anaerobic base, core stability, mental readiness. These have sport set to "universal".
- **Sport-specific nodes (levels 3–6):** ~6–12 nodes covering the sport's key progressions. These have sport set to "${sport}".
- Exactly one "Peak Game Performance" terminal node at level 6.
- Prerequisite edges must form a valid DAG (no cycles).
- Sport-specific nodes at levels 3–4 should depend on relevant universal nodes (e.g., a throwing skill depends on core stability).
- Higher-level sport nodes depend on lower-level sport nodes AND relevant universal nodes.
- Node IDs must be kebab-case (e.g., "ball-handling", "sleep-hygiene").
- Every skill must include a "summary" field: one concise sentence (~15–25 words) describing what the skill involves and why it matters to an athlete's development. Plain language, no coach-speak filler.
- Every skill must include a "diagnosticPrompt" field, but the rules differ by level:
  - **Levels 1–4 (foundation + development):** ONE concrete, on-the-spot drill (~10–30 words) the coach can grade pass / partial / fail in **≤60 seconds, right now, during onboarding**. Must require no prior warmup context, no teammates prepped, no sustained bout. Observable, benchmarkable, and bounded.
    - Good: "3 reps: one-hop throw to a cone 20 ft away - partner counts clean catches (2 of 3)."
    - Good: "Hold a tall plank for 30s - flag hip drop or lumbar sag."
    - Good: "5 swings off a tee; 4 of 5 must line-drive to center net."
    - Bad: "Play a full 7-inning scrimmage." (not on-the-spot, takes hours)
    - Bad: "Demonstrate consistency under pressure over a match." (not observable in 60s)
    - Bad: "Show good form on lifts across a week." (not a single observable moment)
  - **Levels 5–6 (integration / peak game performance):** These integration skills CANNOT be graded in a single on-the-spot task during onboarding - they can only be observed across real competition. For these, set the diagnosticPrompt to an empty string (""). The onboarding engine will infer their status from prerequisite coverage; do not invent a drill.

### Athlete Generation Rules
- Generate exactly 10 athletes, ages 14–17, school years from Freshman through Senior.
- Positions must be drawn from ${sport}'s actual position set.
- Mastery sets must be prerequisite-consistent: if an athlete has mastered node X, they must also have mastered ALL of X's prerequisites.
- Spread athletes across development stages: some early (few masteries), some mid-development, some advanced, some near peak.
- Each athlete needs an id (kebab-case of their name), displayName, firstName, age, position, schoolYear, sport (set to "${sport}"), avatarColor (a valid hex color like "#2563eb"), tagline (one sentence about their development), mastery (array of skill IDs they've mastered), and readiness (0–100 integer).

### Task/Drill Generation Rules - per-skill pools
- Tasks are generated PER SKILL, not as a global list. For EVERY skill in the graph (universal and sport-specific), produce 3–5 tasks whose primary skill is that skill.
- Each task's skillId must match a skill you produced. Each task advances ONE primary skill.
- Within each skill's task pool, xp integers MUST sum to exactly 100. A skill is mastered once the athlete has accumulated 100 XP worth of completed tasks on it.
- **XP-time rule:** 1 XP ≈ 1 minute of fully-focused, fully-productive work for an average serious student. Size xp so the implied duration matches what the task actually takes (a 5-minute drill = 5 XP; a 25-minute situational scrimmage = 25 XP). Tasks should fall in the 5–30 XP range so no single task dominates a session.
- Tasks are graded across the corpus to cover the 10 pedagogy techniques: "Knowledge graph", "Physical frontier", "Expert tutor / autoregulation", "Objective readiness", "Spaced repetition", "Interleaving", "Testing effect", "Non-interference", "Automaticity", "Encompassings". Use each technique at least once across the full task set; a single skill's pool will typically mix 2–4 techniques.

### Task Specificity Rules (CRITICAL - read carefully)
A 16-year-old should be able to read **only** the task and execute it without asking a coach a single follow-up question. Vague guidance is a failure. Generic advice is a failure. A task must be a **prescription**, not a category.

Every task's \`title\` and \`detail\` must answer ALL of the following that apply:
- **Exact dose:** sets × reps × load × distance × duration × tempo × pace × HR zone, whichever fit the modality. Numbers, not adjectives.
- **Equipment / setup:** what they need (cones, foam roller, stopwatch, partner, gym, field, kitchen scale, food log app, etc.) and how to arrange it.
- **Execution cues:** the 1–3 form/technical cues the athlete must hit on each rep.
- **Measurable success criterion:** the pass/fail or score-on-this-rep test that tells the athlete the work counted (e.g., "4 of 5 line drives to center net", "RPE ≤ 7", "≥ 35g protein logged within 60 min of finishing").
- **Logging:** what they enter back into the app or a notebook (rep count, time, weight, food entry, video, RPE).
- **Rest / structure:** rest between sets, total session length, work:rest ratio when relevant.

Title rules:
- The \`title\` must be a concrete, prescriptive headline including a key number when possible (load, reps, distance, time, pace, grams, hours).
- NEVER use a bare category as the title ("Nutrition", "Sleep", "Eat food", "Mobility", "Practice shooting", "Cardio", "Stretch"). These are categories, not tasks, and are an automatic failure.

Detail rules:
- \`detail\` is 2–4 sentences (~30–80 words). Pack in dose + setup + cues + success criterion + logging. Brevity is fine; vagueness is not.
- If the task is a habit/lifestyle task (sleep, nutrition, hydration, recovery), it must still be fully prescriptive: target macros in grams, target hours of sleep, target oz of water, exact timing windows relative to practice, and what to log.

Rationale rules:
- \`rationale\` is one short sentence connecting this specific task to the athlete's progress on the parent skill (e.g., "Closes your aerobic-base gap; +12 XP toward mastering Aerobic base.").

Examples - APPLY THIS BAR TO EVERY TASK:

GOOD title + detail (nutrition, universal "macronutrient-tracking" skill):
- title: "Post-practice protein hit: 35g within 60 min"
- detail: "Within 60 minutes of finishing today's session, eat or drink 35g of protein and 60g of carbs. Examples: 1 scoop whey + 1 banana + 1 cup oats, OR 6oz grilled chicken + 1.5 cups rice. Log the meal in MyFitnessPal under 'Post-practice'. Success = total protein ≥ 35g and timing ≤ 60 min from session end."

GOOD title + detail (basketball "ball-handling"):
- title: "Tennis-ball dribble combo: 4 sets × 60s per hand"
- detail: "Set up at the free-throw line. Dribble a basketball with your strong hand while tossing a tennis ball up and catching it with your weak hand for 60s, then switch. 4 sets total, 30s rest between sets. Cue: eyes up, dribble below knee, no fumbles. Success = ≥ 50 clean tennis-ball catches per 60s set; log highest catch count."

GOOD title + detail (sleep, universal "sleep-hygiene"):
- title: "8h sleep with 22:30 lights-out, screens off by 22:00"
- detail: "Tonight: phone in another room by 22:00, in bed reading (paper) by 22:15, lights-out by 22:30, alarm at 06:30 (8h target). Log actual sleep/wake times tomorrow morning in the app. Success = ≥ 7h45m time-in-bed AND no screens after 22:00."

BAD examples (DO NOT PRODUCE - automatic failure):
- title: "Eat food"  - category, no dose, no timing, no logging.
- title: "Get good sleep"  - no target hours, no protocol, not measurable.
- title: "Practice shooting"  - no rep count, no spot, no success criterion.
- title: "Mobility work"  - no specific drills, no duration, no cues.
- detail: "Make sure to fuel properly after training."  - empty, generic, ignorable.
- detail: "Work on your handles."  - no dose, no cue, no measurable outcome.

Each task needs:
- \`id\` (kebab-case; include the skillId for uniqueness)
- \`shortLabel\` (2–3 words; the dashboard chip - keep it tight, e.g. "35g protein hit", "60s dribble combo", "8h sleep")
- \`title\` (concrete prescription, see rules above; usually 6–12 words and includes a number)
- \`detail\` (2–4 sentences, ~30–80 words; full protocol per rules above)
- \`sport\` (either "universal" or "${sport}")
- \`technique\` (one of the 10 above)
- \`skillId\` (the primary skill this task advances)
- \`xp\` (integer 1–100; per the XP-time rule)
- \`rationale\` (one sentence linking task → skill progress per rules above)

### Skill Short Labels
- Provide a skillShortLabels object mapping every skill ID to a 1–2 word abbreviation for use in heatmap column headers.

## Coach Requirements (team-wide)
${requirements || 'No specific requirements provided.'}

## Iteration Behavior
When the user asks to adjust the graph, preserve unchanged portions and modify only what they asked about. Explain what changed and why in your chatReply.

## Output Format
Return a JSON object with:
- chatReply: A short 1–2 sentence summary of the plan. DO NOT include per-level bullet lists, node counts, or long explanations. Keep it under ~40 words.
- graph: The complete graph object with skills, athletes, tasks, and skillShortLabels.`
}

export interface EscalationSkill {
  id: string
  label: string
  summary?: string
  level: number
  prereqs: string[]
  prereqLabels?: string[]
}

export interface EscalationPriorAttempt {
  prompt: string
  verdict: 'pass' | 'fail' | 'conditional'
  skillLabel?: string
}

/**
 * System prompt for an escalation probe. The diagnostic engine asks the AI
 * to invent a SINGLE on-the-spot drill that's strictly harder than every
 * prior pass / partial attempt for this branch. Used when the normal probe
 * pass ended without a fail, so we keep climbing until we find a ceiling.
 */
export function buildEscalationProbePrompt(
  sport: string,
  skill: EscalationSkill,
  priorAttempts: EscalationPriorAttempt[],
  athleteContext: AthleteContext | undefined,
): string {
  const name = athleteContext?.firstName || athleteContext?.name || 'this athlete'
  const profileLines: string[] = []
  if (athleteContext?.position) profileLines.push(`- Position: ${athleteContext.position}`)
  if (athleteContext?.schoolYear) profileLines.push(`- School year: ${athleteContext.schoolYear}`)
  if (typeof athleteContext?.age === 'number') profileLines.push(`- Age: ${athleteContext.age}`)
  if (athleteContext?.tagline) profileLines.push(`- Tagline: ${athleteContext.tagline}`)
  if (typeof athleteContext?.readiness === 'number') {
    profileLines.push(`- Readiness: ${athleteContext.readiness}/100`)
  }
  const prereqLine =
    skill.prereqLabels && skill.prereqLabels.length > 0
      ? skill.prereqLabels.join(', ')
      : skill.prereqs.join(', ')

  const attemptLines = priorAttempts.length === 0
    ? '(none yet)'
    : priorAttempts
        .map((a) => {
          const tag = a.verdict === 'pass' ? 'PASSED' : a.verdict === 'conditional' ? 'PARTIAL' : 'FAILED'
          const who = a.skillLabel ? `${a.skillLabel}: ` : ''
          return `- ${who}"${a.prompt}" → ${tag}`
        })
        .join('\n')

  // Keep this prompt SHORT. The output is ~50 tokens; a long system
  // prompt dominates time-to-first-token. The athlete profile is one
  // compact line, prior attempts are kept terse, and the rules are
  // boiled down to the must-haves only.
  const profileLine = profileLines.length > 0
    ? profileLines.map((l) => l.replace(/^- /, '')).join(', ')
    : 'no profile'

  return `Design ONE harder ${sport} probe for ${name} (${profileLine}).

Skill: ${skill.label} (Lvl ${skill.level})${skill.summary ? ` - ${skill.summary}` : ''}
Prereqs: ${prereqLine || '(none)'}

Prior attempts on this branch (escalate beyond every passed/partial one):
${attemptLines}

Rules:
- ONE drill, ~15-25 words, gradeable pass/partial/fail in ≤60s right now.
- Strictly harder than every passed/partial attempt above (more reps, faster, tighter benchmark, harder variant, less rest, added constraint).
- Specific to "${skill.label}" at level ${skill.level}. For higher-level integration skills, use a compressed constraint micro-drill (e.g. "3v0 break, 8s shot clock, 2 of 3 finishes at the rim").
- Include the exact pass/fail benchmark in the drill text.
- No "play a full match", no multi-day asks, no "demonstrate consistency".

Output JSON: { prompt: "<drill text>", rationale: "<one short sentence: what this escalates>" }`
}

export function buildAthleteSystemPrompt(
  sport: string,
  requirements: string,
  athleteContext: AthleteContext | undefined,
): string {
  const name = athleteContext?.name ?? 'this athlete'
  const profileLines: string[] = []
  if (athleteContext?.position) profileLines.push(`- Position: ${athleteContext.position}`)
  if (athleteContext?.schoolYear) profileLines.push(`- School year: ${athleteContext.schoolYear}`)
  if (typeof athleteContext?.age === 'number') profileLines.push(`- Age: ${athleteContext.age}`)
  if (athleteContext?.tagline) profileLines.push(`- Tagline: ${athleteContext.tagline}`)
  if (typeof athleteContext?.readiness === 'number') profileLines.push(`- Readiness: ${athleteContext.readiness}/100`)
  if (athleteContext?.masteredIds && athleteContext.masteredIds.length > 0) {
    profileLines.push(`- Already mastered: ${athleteContext.masteredIds.join(', ')}`)
  }
  const profileBlock = profileLines.length > 0 ? profileLines.join('\n') : '- No profile details supplied.'

  return `You are an expert athletic development coach fine-tuning an individual ${sport} athlete's plan on top of an existing TEAM BASELINE.

## Athlete profile for ${name}
${profileBlock}

## Coach's adjustments for ${name}
${requirements || 'No specific adjustments provided; follow chat messages.'}

## Editing Rules
- You will be given the full team baseline graph ([Team baseline graph: ...]) AND the current in-progress resolved graph for this athlete ([Current athlete graph: ...]).
- Prefer MINIMAL, TARGETED changes to the team baseline. Only add, remove, or modify skills that are meaningfully different for ${name}.
- Do NOT rewrite unrelated portions of the graph. Keep team baseline nodes intact unless the coach explicitly wants them changed.
- Preserve the overall tier structure and DAG validity.
- Node IDs must remain kebab-case. New athlete-specific nodes should use descriptive, unique kebab-case ids (e.g., "elbow-reinforcement").
- Every skill must include a "summary" field: one concise sentence (~15–25 words) describing what the skill involves and why it matters. Carry existing summaries through unchanged for unchanged skills; write fresh summaries for any new athlete-specific nodes.
- Return the FULL RESOLVED graph for this athlete (team baseline + all athlete-specific adjustments merged). The client will diff against the baseline to compute the athlete's delta.

## Athletes & Tasks
- Keep the \`athletes\` array identical to the team baseline (copy it through unchanged).
- Keep the \`tasks\` array identical to the team baseline (copy it through unchanged).
- Update \`skillShortLabels\` to include entries for any new skills you add.

## Iteration Behavior
When the user iterates, preserve all unchanged portions. Explain in your chatReply what specifically diverges from the team baseline for ${name} and why.

## Output Format
Return a JSON object with:
- chatReply: A short 1–2 sentence summary of what diverges from the team baseline for ${name}. DO NOT include bullet lists or long explanations. Keep it under ~40 words.
- graph: The full resolved graph object (skills + athletes + tasks + skillShortLabels).`
}
